lib/deck.rb

```diff
@@ -0,0 +1,40 @@
+class Deck
+  attr_reader :cards
+
+  def initialize(cards)
+    @cards = cards
+  end
+
+  def rank_of_card_at(index)
+    cards[index].rank
```

Comment:
The rule of thumb is we want to use the instance variable in the class where it's defined (rather than the method). Both ways work, it's just convention to use the instance var.
```suggestion
    @cards[index].rank
```